src/report/core/sqs-report-calculator.service.ts

```diff
@@ -1,23 +1,57 @@
-import { Injectable } from '@nestjs/common';
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
 import { ReportCalculator } from './report-calculator';
 import { ReportData } from '../../domain/report-data';
-import { delay, Observable, of } from 'rxjs';
+import { map, mergeAll, Observable, switchMap } from 'rxjs';
 import { ReportCalculation } from '../../domain/report-calculation';
-import { Reference } from '../../domain/reference';
+import { DefaultReportStorage } from '../storage/report-storage.service';
+import { CouchSqsClient } from '../../couchdb/couch-sqs.client';
 import { v4 as uuidv4 } from 'uuid';
+import { Reference } from '../../domain/reference';
 
 @Injectable()
 export class SqsReportCalculator implements ReportCalculator {
+  constructor(
+    private sqsClient: CouchSqsClient,
+    private reportStorage: DefaultReportStorage,
+  ) {}
+
   calculate(reportCalculation: ReportCalculation): Observable<ReportData> {
-    return of(
-      new ReportData(
-        `ReportData:${uuidv4()}`,
-        reportCalculation.report,
-        new Reference(reportCalculation.id),
-      ).setData({
-        foo: 'bar',
-        dummyReportData: 'foo',
+    return this.reportStorage.fetchReport(reportCalculation.report).pipe(
+      switchMap((report) => {
+        if (!report) {
+          throw new NotFoundException();
+        }
+
+        if (report.mode !== 'sql') {
+          throw new BadRequestException();
+        }
+
+        if (report.queries.length === 0) {
+          throw new BadRequestException();
```

Comment:
maybe InternalServerError or some other 50x? --> queries are parsed from the ReportConfig, which is administered by us manually